open-banking-accelerator/internal-apis/internal-webapps/com.wso2.openbanking.accelerator.event.notifications.endpoint/src/main/java/com/wso2/openbanking/accelerator/event/notifications/endpoint/api/EventSubscriptionEndpoint.java

```diff
@@ -0,0 +1,257 @@
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * Events Notification Subscription API.
+ */
+@Path("/subscription")
+public class EventSubscriptionEndpoint {
+    private static final Log log = LogFactory.getLog(EventSubscriptionEndpoint.class);
+
+    private static final EventSubscriptionServiceHandler eventSubscriptionServiceHandler = EventSubscriptionUtils.
+            getEventSubscriptionServiceHandler();
+
+    /**
+     * Register an Event Notification Subscription.
+     */
+    @SuppressFBWarnings({"JAXRS_ENDPOINT", "SERVLET_HEADER"})
+    @POST
+    @Path("/")
+    @Consumes({"application/json; charset=utf-8"})
+    @Produces({"application/json; charset=utf-8"})
+    @ApiOperation(value = "Create Subscriptions", tags = {" Create Subscriptions"})
+    public Response registerSubscription(@Context HttpServletRequest request, @Context HttpServletResponse response) {
+
+        EventSubscriptionDTO eventSubscriptionDTO = new EventSubscriptionDTO();
+
+        //check if the client id is present in the header
+        String clientId = request.getHeader(EventNotificationEndPointConstants.X_WSO2_CLIENT_ID);
+        if (StringUtils.isBlank(clientId)) {
+            return Response.status(Response.Status.BAD_REQUEST).entity(EventNotificationUtils.
+                    getErrorDTO(EventNotificationEndPointConstants.MISSING_REQUEST_HEADER,
+                            EventNotificationConstants.MISSING_HEADER_PARAM_CLIENT_ID)).build();
+        }
+        eventSubscriptionDTO.setClientId(request.getHeader(EventNotificationEndPointConstants.X_WSO2_CLIENT_ID));
+
+        // extract the payload from the request
+        try {
+            JSONObject requestData = EventSubscriptionUtils.getJSONObjectPayload(request);
+            if (requestData != null) {
+                eventSubscriptionDTO.setRequestData(requestData);
+            } else {
+                log.error("Subscription request payload is missing");
+                return Response.status(Response.Status.BAD_REQUEST).
+                        entity(EventNotificationUtils.
+                                getErrorDTO(EventNotificationEndPointConstants.MISSING_REQUEST_PAYLOAD,
+                                        EventNotificationEndPointConstants.MISSING_JSON_REQUEST_PAYLOAD)).build();
+            }
+        } catch (IOException e) {
+            log.error("Invalid Payload received", e);
+            return Response.status(Response.Status.BAD_REQUEST).
+                    entity(EventNotificationUtils.
+                            getErrorDTO(EventNotificationEndPointConstants.INVALID_REQUEST_PAYLOAD,
+                                    EventNotificationEndPointConstants.REQUEST_PAYLOAD_ERROR)).build();
+        } catch (ParseException e) {
```

Comment:
exception not logged

Reply:
fixed with ea8fc6ef449ca5c80e471dcf3749615b21da0d39